File: 卡码/代码随想录/Go/二叉树/06-二叉搜索树修改与构造/LeetCode_0108.go
//108. 将有序数组转换为二叉搜索树
package main

import (
	. "binaryTree/types"
)

func traversal(nums []int,left,right int) *TreeNode{
	if left>right {
		return nil
	}
	mid:=(left+right)/2
	root:=&TreeNode{nums[mid],nil,nil}
	root.Left=traversal(nums,left,mid-1)
	root.Right=traversal(nums,mid+1,right)
	return root
}

func sortedArrayToBST(nums []int) *TreeNode {
	return traversal(nums,0,len(nums)-1)
}

File: 卡码/代码随想录/JS/二叉树/06-二叉搜索树修改与构造/LeetCode_0108.js
//108. 将有序数组转换为二叉搜索树
import TreeNode from "../TreeNode.js"

const traversal=(nums,left,right)=>{
  if(left>right) return null;
  const mid=Math.floor((left+right)/2);
  const root=new TreeNode(nums[mid]);
  root.left=traversal(nums,left,mid-1);
  root.right=traversal(nums,mid+1,right);
  return root;
};

const sortedArrayToBST = function(nums) {
  return traversal(nums,0,nums.length-1);
};